services/etl/provenance.py

```diff
@@ -0,0 +1,279 @@
+"""
+Generate extracts a set of entities from the model and corresponding
+commit relevant to the ETL conventions document. The set of entities
+consist of ETL conventions data for the model, tables, and fields as
+well as provenance information including causual or associated events
+and the people or systems that were involved or are responsible.
+
+All information produced here is stateless since historical information
+is not available.
+
+The structure of an entity is as follows:
+
+- `domain` - The top-level domain the entity should be written to.
+- `ident` - The unique identity value of the entity. Entities that do
+not (or should not) be referenced by other entities do require this
```

Comment:
Thanks. It should state why it should be set rather than why not.

> An identity should be provided if the entity is a continuant in the domain so all occurrents can be associated.